src/Microsoft.AspNetCore.Sockets.Client/Connection.cs

```diff
@@ -166,18 +173,36 @@ public Task StartAsync(ITransport transport, HttpClient httpClient)
         {
             try
             {
-                _logger.LogTrace("Beginning receive loop");
+                _logger.LogTrace("Beginning receive loop.");
 
                 while (await Input.WaitToReadAsync())
                 {
+                    if (_connectionState != ConnectionState.Connected)
```

Comment:
As per this check if the data is received after the connection moved to the disconnected state we will reject it - this can only happen when the user is stopping the connection and they basically said they don't want to receive messages. If state changed after this check you received the data before you moved to the disconnected state. Firing the event seems to be fine in that case.
From the functional perspective the behavior I wanted was - receive event should not be scheduled after close event and receive should not be scheduled after DisposeAsync exits even if the invocation is not awaited.